src/include/rbd_types.h

```diff
@@ -106,4 +107,9 @@ struct rbd_obj_header_ondisk {
 	struct rbd_obj_snap_ondisk snaps[0];
 } __attribute__((packed));
 
+enum group_image_link_state {
+	LINK_NORMAL,
```

Comment:
Yeah, I thought about it again earlier today and agree it should just be attached and "incomplete". I would still like to see the _add and _clean methods just be combined into a _set method where the enum is provided.